### PR TITLE
[T4A3][T10-B2] Low Yong Cheng

### PR DESCRIPTION
**Pull request by yongchenglow**
Created a parent class Contact for phone, email and address

### CODE REVIEW

**Comment by yongchenglow** [comment]
Ready for Review

File: src/seedu/addressbook/data/person/Address.java [hunk 1]
```diff
@@ -6,26 +6,22 @@
...
     /**
      * Validates given address.
      *
      * @throws IllegalValueException if given address string is invalid.
      */
     public Address(String address, boolean isPrivate) throws IllegalValueException {
-        this.isPrivate = isPrivate;
-        if (!isValidAddress(address)) {
+    	super(address, isPrivate);
+    	if (!isValidAddress(address)) {
```
Comment (nirandiw) [inline src/seedu/addressbook/data/person/Address.java:22]: Can push the validation to the super constructor too, if you want

File: src/seedu/addressbook/data/person/Contact.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
+package seedu.addressbook.data.person;
+
+import seedu.addressbook.data.exception.IllegalValueException;
+
+/**
+ * Represents a Person's Contact.
+ * Parent class for Address, Phone and Email
```
Comment (nirandiw) [inline src/seedu/addressbook/data/person/Contact.java:7]: Don't include the subclasses in a super class header comment. You never know who will use this super class in future.

File: src/seedu/addressbook/data/person/Contact.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    }
+
+    @Override
+    public String toString() {
+        return value;
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        return other == this // short circuit if same object
```
Comment (nirandiw) [inline src/seedu/addressbook/data/person/Contact.java:30]: At any point if you decide to override the equals method in sub classes, keep in mind to override the hashCode() too.

**Comment by nirandiw** [comment]
@yongchenglow Some comments added. Please close the PR after reading comments.

**Pull request closed without merging**